bowler/query.py

```diff
@@ -82,11 +82,16 @@ def wrapped(self: Q, *args, **kwargs) -> Q:
 
 
 class Query:
-    def __init__(self, *paths: Union[str, List[str]]) -> None:
+    def __init__(
+        self,
+        *paths: Union[str, List[str]],
+        filename_matcher: Callable[[str], bool] = None,
```

Comment:
Seems like a good case for a custom type in `bowler/types.py`, something like `FilenameMatcher` maybe, and have it take the `Filename` type while we're at it?